src/mlang/m_frontend/mast_to_mir.ml

```diff
@@ -1216,10 +1216,16 @@ let add_dummy_definitions_for_variable_declarations (var_data : Mir.variable_dat
               Mir.VariableMap.add var
                 {
                   Mir.var_definition =
-                    Mir.TableVar
-                      ( size,
-                        Mir.IndexGeneric
-                          (Pos.same_pos_as (Mir.Literal Mir.Undefined) var.Mir.Variable.name) );
+                    begin
+                      match io with
+                      | Mir.Input -> Mir.InputVar
+                      | _ ->
+                          Mir.TableVar
+                            ( size,
+                              Mir.IndexGeneric
+                                (Pos.same_pos_as (Mir.Literal Mir.Undefined) var.Mir.Variable.name)
+                            )
+                    end;
```

Comment:
This should be refactored into a `Mir.create_undefined_value : io -> value` function.

Reply:
`variable_def` rather than `value`, no?

Reply:
As discussed elsewhere, there is limited factoring to be done here, so I'll leave this as is.